walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,87 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors; 
+import java.util.stream.Stream; 
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+    int dayStart = TimeRange.START_OF_DAY;
+    int dayEnd = TimeRange.END_OF_DAY;
+    int start = dayStart;
+    int end;
+    TimeRange previousTime = TimeRange.fromStartDuration(0,0);
+    
+    List<TimeRange> sortedTimeRanges = getAllAttendeeRanges(events, request);
+    ArrayList<TimeRange> possibleTimeRanges = new ArrayList<>();
+
+    //Returns an empty List should the requested duration be longer than a total day.
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) return possibleTimeRanges;
+
+    //For-each that iterates through sortedTimeRanges to determine available times for request's duration.
+    for (TimeRange timeRange : sortedTimeRanges) {
+        if (timeRange.start() == dayStart) start = timeRange.end();
+
```

Comment:
pls remove the unnecessary blank lines.

Reply:
Done. Let me know If there are more lines that need to be removed/added.